### PR TITLE
Cleanup after model is deleted

### LINKED ISSUE #12: Translations not deleted automatically once related "translatable" entity is deleted
### Winter CMS Build

dev-develop

### PHP Version

8.0

### Database engine

MySQL/MariaDB

### Plugins installed

_No response_

### Issue description

When a translatable model gets deleted (physically from the database) no related translations of that model get deleted automatically from winter_translate_attributes and winter_translate_indexes.

### Steps to replicate

1. Create a translatable model
2. Save it to the database
3. Delete it

Expected result:
All related translations should be removed

Actual result:
Translations remain in winter_translate_attributes and winter_translate_indexes

### Workaround

_No response_

### Any further details?

_No response_

### PR DESCRIPTION
**Pull request by mjauvin**
Fixes #12

### CODE REVIEW

**Comment by mjauvin** [comment]
@lex0r can you test if this resolves the issue?

**Comment by LukeTowers** [comment]
How does this handle the parent record being soft deleted?

**Comment by mjauvin** [comment]
It doesn't... I'll apply conditionally ij afterModelDelete() then.

**Comment by mjauvin** [comment]
@LukeTowers that should address the softdelete concern.

**Comment by LukeTowers** [comment]
Is there a way that we can support deleting the data when a softdeleted record is permanently deleted?

File: behaviors/TranslatableModel.php [hunk 1]
```diff
@@ -26,8 +26,32 @@ public function __construct($model)
...
-            'name' => 'model'
+            'name' => 'model',
         ];
+
+        $this->model->bindEvent('model.afterDelete', [$this, 'afterModelDelete']);
+    }
+
+    public function afterModelDelete()
+    {
+        if (method_exists($this->model, 'isSoftDelete') && $this->model->isSoftDelete()) {
```
[the remark this answers is not in the document]
Comment (mjauvin) [reply behaviors/TranslatableModel.php]: Fixed!

**Review by LukeTowers — APPROVED** [review]

**Comment by LukeTowers** [comment]
Just waiting on @lex0r to test and confirm and this should be good to merge. 

Thanks @mjauvin, great job!

**Comment by mjauvin** [comment]
@lex0r it would really be appreciated if you could test this in your environment so we can get this merged.

Thanks!

**Comment by lex0r** [comment]
Hi @mjauvin 
Sorry for the delay, it works for me and thanks for the fix!

**Pull request merged**